pkg/driver/node_test.go

```diff
@@ -1183,6 +1183,57 @@ func TestGetVolumesLimit(t *testing.T) {
 				return m
 			},
 		},
+		// 1 gpu
+		{
+			name: "g4dn.xlarge_volume_attach_limit",
+			options: &Options{
+				VolumeAttachLimit:         -1,
+				ReservedVolumeAttachments: -1,
+			},
+			expectedVal: 24,
+			metadataMock: func(ctrl *gomock.Controller) *metadata.MockMetadataService {
+				m := metadata.NewMockMetadataService(ctrl)
+				m.EXPECT().GetRegion().Return("us-west-2")
+				m.EXPECT().GetInstanceType().Return("g4dn.xlarge")
+				m.EXPECT().GetNumBlockDeviceMappings().Return(0)
+				m.EXPECT().GetNumAttachedENIs().Return(1)
+				return m
+			},
+		},
+		// 1 gpu
+		{
+			name: "g4ad.xlarge_volume_attach_limit",
+			options: &Options{
+				VolumeAttachLimit:         -1,
+				ReservedVolumeAttachments: -1,
+			},
+			expectedVal: 24,
+			metadataMock: func(ctrl *gomock.Controller) *metadata.MockMetadataService {
+				m := metadata.NewMockMetadataService(ctrl)
+				m.EXPECT().GetRegion().Return("us-west-2")
+				m.EXPECT().GetInstanceType().Return("g4ad.xlarge")
+				m.EXPECT().GetNumBlockDeviceMappings().Return(0)
+				m.EXPECT().GetNumAttachedENIs().Return(1)
+				return m
+			},
+		},
+		// 4 gpus
```

Comment:
Nit: Would it be helpful to append this comment to the test name itself `g4dn.12xlarge_volume_attach_limit (4 GPUs, 1 InstanceStoreVolume)`, and adding the number of Instance Store Volumes instead of commenting? 

I'm thinking that this would be helpful if these unit tests fail. Developers can easily see test name for the test's assumptions, instead of stumbling across this comment. Furthermore the inclusion of InstanceStoreVolumes makes clear that GPUs aren't the only thing that uses up slot.

Reply:
agree will add in next rev